gram.y

```diff
@@ -218,8 +218,8 @@ ifacehead	: T_INTERFACE name
 			}
 
 			iface_init_defaults(iface);
-			strncpy(iface->props.name, $2, IFNAMSIZ-1);
-			iface->props.name[IFNAMSIZ-1] = '\0';
+			memset(&iface->props.name, 0, sizeof(iface->props.name));
```

Comment:
FWIW: _"looks strange"_

_"Strange"_ as in none of the previous _hugs_ did add `memset()`.

Reply:
I'm not 100% sure the `memset` is required, except in cases where the variable is re-used, and we might leak part of what the old content used to be.
```
char s[16] = "longstring";
// s actually contains "longstring\0\0\0\0\0\0"
strlcpy(s, "foo", sizeof(s));
// s actually contains "foo\0string\0\0\0\0\0\0"
```